SQLQuery_Week09_Homework.sql

```diff
@@ -0,0 +1,414 @@
+
+--OR
+
+-- GO
+-- CREATE PROCEDURE GetExperienceByYearsForPresentLibrarians(
+--     @yearsExperience INT
+-- )
+-- AS 
+-- BEGIN
+--     SELECT *
+--     FROM Librarians l
+--     WHERE l.IsOnHoliday = 0 AND DATEDIFF(YYYY,l.HireDate,GETDATE()) > @yearsExperience;
+-- END
+-- GO
+
+-- EXEC GetExperienceByYearsForPresentLibrarians 5;
+
+-- 7. Create a stored procedure that displays, based on the book title, the number of libraries in which the book is found. Call the procedure with three different book titles.
+
+GO
+CREATE PROCEDURE GetNumberOfLibrariesByBookTitle(
+    @customerBookTitle VARCHAR(100)
+)
+AS 
+BEGIN
+    SELECT COUNT(l.Id)
+    FROM Books_Libraries bl
+        INNER JOIN Books b ON bl.BookId = b.Id
+        INNER JOIN Libraries l ON bl.LibraryId = l.Id
+    GROUP BY b.Title
+    HAVING b.Title = @customerBookTitle;
+END
+GO
+
+EXEC GetNumberOfLibrariesByBookTitle 'Book1';
+EXEC GetNumberOfLibrariesByBookTitle 'Book3';
+EXEC GetNumberOfLibrariesByBookTitle 'Book5';
+
+-- 8. Create a function that returns, based on the author's email, the number of copies sold for each of his / her books. Call the function with three different emails.
+
+GO
+CREATE VIEW BookIdAndNumberOfCopiesByAuthorId
+AS
+    SELECT b.Title, s.LibraryId, s.NumberOfCopies, b.AuthorId 
+    FROM Books_Sales bs
+     INNER JOIN Books b ON b.Id = bs.BookId
+     INNER JOIN Sales s ON s.Id = bs.SalesId;
+GO
+
+GO
+CREATE FUNCTION dbo.GetNumberOfCopiesOfBooksByAuthorsEmail(
+    @customerAuthorsEmail VARCHAR(100)
+)
+RETURNS INT
+AS
+BEGIN
+    DECLARE @numberOfCopies INT;
+    SET @numberOfCopies = (SELECT SUM(bc.NumberOfCopies)
+                           FROM BookIdAndNumberOfCopiesByAuthorId bc
+                                INNER JOIN Authors a ON a.Id = bc.AuthorId
+                            GROUP BY a.Email
+                            HAVING a.Email = @customerAuthorsEmail);
+    RETURN @numberOfCopies;
+END;
+GO
+
+SELECT dbo.GetNumberOfCopiesOfBooksByAuthorsEmail('mp@email.com') as 'Number of copies';
+SELECT dbo.GetNumberOfCopiesOfBooksByAuthorsEmail('ms@email.com') as 'Number of copies';
+SELECT dbo.GetNumberOfCopiesOfBooksByAuthorsEmail('cz@email.com') as 'Number of copies';
+
+
+-- 9. Create a trigger so that when a new librarian is added to the system with no library assigned (NULL on LibraryId), he / she is automatically assigned to a library.
+
+GO
+CREATE VIEW LibrariesAndLibrarians
+AS
+    SELECT LL.Id 'Library ID', LL.Name 'Library Name', L.Id 'Librarian ID', L.Name 'Librarian Name'
+    FROM Sales s
+        INNER JOIN Librarians L ON s.LibrarianId = L.Id
+        INNER JOIN Libraries LL ON s.LibraryId = LL.Id;
+GO
+
+SELECT * FROM LibrariesAndLibrarians
+
+GO
+CREATE TRIGGER AutoAssignedLibraryToNewLibrarianWithoutLibrary
+ON Librarians
+AFTER INSERT
+AS
+BEGIN
+    INSERT INTO Sales (NumberOfCopies, LibraryId, LibrarianId)
+    VALUES (0, 1, (SELECT MAX(Id) FROM Librarians))
+END
```

Comment:
I have to admit I'm a bit lost with the logic here. It doesn't seem to solve the requirement. Please have a look on my example once it's published.